src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -1439,6 +1440,7 @@ def download(cls, pretrained_model_name, **kwargs) -> Union[str, os.PathLike]:
                 if (
                     len(safetensors_variant_filenames) > 0
                     and safetensors_model_filenames != safetensors_variant_filenames
+                    and not is_sharded
```

Comment:
Sharded checkpoints with a variant ("fp16") for example may have lesser shards actually. Consider https://huggingface.co/fal/AuraFlow/tree/main/transformer, for example. 

For non-fp16, it has three shards and for fp16 it has two.